src/index.js

```diff
@@ -83,14 +86,28 @@ const extendCodeBlockWithSyntaxHighlighting = ( settings ) => {
 						</PanelRow>
 					</PanelBody>
 				</InspectorControls>
-				<div key="editor-wrapper" className={ className }>
-					<PlainText
-						value={ attributes.content }
-						onChange={ ( content ) => setAttributes( { content } ) }
-						placeholder={ __( 'Write code…', 'syntax-highlighting-code-block' ) }
-						aria-label={ __( 'Code', 'syntax-highlighting-code-block' ) }
-					/>
-				</div>
+				{
+					useLightBlockWrapper ?
+						// This must be kept in sync with <https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/code/edit.js>.
+						<BlockEditor.__experimentalBlock.pre>
+							<PlainText
+								__experimentalVersion={ 2 }
+								tagName="code"
+								value={ attributes.content }
+								onChange={ ( content ) => setAttributes( { content } ) }
+								placeholder={ __( 'Write code…', 'syntax-highlighting-code-block' ) }
+								aria-label={ __( 'Code', 'syntax-highlighting-code-block' ) }
+							/>
+						</BlockEditor.__experimentalBlock.pre> :
+						<div key="editor-wrapper" className={ className }>
+							<PlainText
+								value={ attributes.content }
+								onChange={ ( content ) => setAttributes( { content } ) }
+								placeholder={ __( 'Write code…', 'syntax-highlighting-code-block' ) }
+								aria-label={ __( 'Code', 'syntax-highlighting-code-block' ) }
```

Comment:
Since these attributes are shared between the two blocks. Would we be able to do something?

```js
const shared_props = {
    value: attributes.content,
    onChange: ( content ) => setAttributes( { content } ),
    placeholder: __( 'Write code…', 'syntax-highlighting-code-block' ),
    'aria-label': __( 'Code', 'syntax-highlighting-code-block' ),
};
```

And then in our JSX

```jsx
<BlockEditor.__experimentalBlock.pre>
    <PlainText
        {...shared_props}
        __experimentalVersion={ 2 }
        tagName="code"
    />
</BlockEditor.__experimentalBlock.pre> :
<div key="editor-wrapper" className={ className }>
    <PlainText {...shared_props} />
</div>
```

Reply:
Good idea. Done in bef1891.